widgets/linechart/formatter.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linechart
+
+import (
+	"fmt"
+	"math"
+	"time"
+)
+
+// durationSingleUnitPrettyFormat returns the pretty format in one single
+// unit for a time.Duration, the different returned unit formats
+// are: nanoseconds, microseconds, milliseconds, seconds, minutes
+// hours, days.
+func durationSingleUnitPrettyFormat(d time.Duration, decimals int) string {
+	// Check if the duration is less than 0.
+	prefix := ""
+	if d < 0 {
+		prefix = "-"
+		d = time.Duration(math.Abs(d.Seconds()) * float64(time.Second))
+	}
+
+	switch {
+	// Nanoseconds.
+	case d.Nanoseconds() < 1000:
+		dfmt := prefix + "%dns"
```

Comment:
This switch reads really well, thanks for using such a well structured approach.

(nit) Go tends to use mixedCaps for multi-word variable names.

Can we call this variable dFmt ?

https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps

Reply:
In that case, shouldn't be `dFMT`? (https://github.com/golang/go/wiki/CodeReviewComments#initialisms)

Reply:
I was actually quite confused between the two for a while. The way I understand it, fmt is an abbreviation, not an initialism. Abbreviations (the word "format" shortened to fmt) follow the standard mixedCaps rules.

Initialisms (each letter stands for its own word, like HTTP) use the second rule.

Either way this is mostly informational as these things are a bit too much on the "extreme style police" side of code reviews.

Reply:
Right! I've always been a little bit confused with this, anyway `dFmt` it's perfect :)